### PR TITLE
Can generate blocks through RPC

### PR DESCRIPTION
**Pull request by NicolasDorier**

### CODE REVIEW

File: Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs [hunk 1]
```diff
@@ -73,6 +74,13 @@ public Task Stop()
                 return new TransactionBriefModel(trx);
         }
 
+        [ActionName("getblockcount")]
+        public int GetBlockCount()
+        {
+            var consensusLoop = this.FullNode.Services.ServiceProvider.GetRequiredService<ConsensusLoop>();
```
Comment (dangershony) [inline Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs]: You could use the ChainState on this.FullNode

File: Stratis.Bitcoin/RPC/Controllers/WalletRPCController.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                get; set;
+            }
+            public HdAccount Account
+            {
+                get;
+                set;
+            }
+        }
+        public WalletRPCController(IServiceProvider serviceProvider, IWalletManager walletManager)
```
Comment (dangershony) [inline Stratis.Bitcoin/RPC/Controllers/WalletRPCController.cs]: FYI PowMining should be in the DI as well (after we fix the registration issue)

Comment (NicolasDorier) [inline Stratis.Bitcoin/RPC/Controllers/WalletRPCController.cs]: This is an optional dependency though

File: Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs (hunk 1 above)
Comment (NicolasDorier) [inline Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs]: I don't have ChainState in IFullNode

File: Stratis.Bitcoin/RPC/Controllers/WalletRPCController.cs (hunk 2 above)
Comment (dangershony) [inline Stratis.Bitcoin/RPC/Controllers/WalletRPCController.cs]: Ah right yes hmmm, not without design changes.

**Pull request merged**